### PR TITLE
Add incompatibility caused by deprecated features in mysql-compatibility.md

### PR DESCRIPTION
**Pull request by Joyinqin**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/4590
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: mysql-compatibility.md [hunk 1]
```diff
@@ -178,4 +178,11 @@ TiDB 支持大部分 [SQL 模式](/sql-mode.md)。不支持的 SQL 模式如下
 
 + 不支持 FLOAT4/FLOAT8。
 
-+ 不支持 `SQL_TSI_*`（包括 `SQL_TSI_MONTH`、`SQL_TSI_WEEK`、`SQL_TSI_DAY`、`SQL_TSI_HOUR`、`SQL_TSI_MINUTE` 和 `SQL_TSI_SECOND`，但不包括 `SQL_TSI_YEAR`）。
\ No newline at end of file
++ 不支持 `SQL_TSI_*`（包括 `SQL_TSI_MONTH`、`SQL_TSI_WEEK`、`SQL_TSI_DAY`、`SQL_TSI_HOUR`、`SQL_TSI_MINUTE` 和 `SQL_TSI_SECOND`，但不包括 `SQL_TSI_YEAR`）。
+
+### MySQL 弃用功能导致的不兼容问题
+
+TiDB 不支持在 MySQL 中标记为弃用的功能。包括：
```
Comment (TomShawn) [inline mysql-compatibility.md:185]:
```suggestion
TiDB 不支持 MySQL 中标记为弃用的功能，包括：
```

File: mysql-compatibility.md [hunk 2]
```diff
@@ -178,4 +178,11 @@ TiDB 支持大部分 [SQL 模式](/sql-mode.md)。不支持的 SQL 模式如下
...
+
+* 指定浮点类型的精度。MySQL 8.0 [弃用](https://dev.mysql.com/doc/refman/8.0/en/floating-point-types.html)了此功能，建议改用 `DECIMAL` 类型。
+* `ZEROFILL` 属性。 MySQL 8.0 [弃用](https://dev.mysql.com/doc/refman/8.0/en/numeric-type-attributes.html)了此功能，建议改在应用程序中填充数字值。
```
Comment (TomShawn) [inline mysql-compatibility.md:188]:
```suggestion
* `ZEROFILL` 属性。 MySQL 8.0 [弃用](https://dev.mysql.com/doc/refman/8.0/en/numeric-type-attributes.html)了此功能，建议改在业务应用中填充数字值。
```

File: mysql-compatibility.md [hunk 3]
```diff
@@ -178,4 +178,11 @@ TiDB 支持大部分 [SQL 模式](/sql-mode.md)。不支持的 SQL 模式如下
 
 + 不支持 FLOAT4/FLOAT8。
 
-+ 不支持 `SQL_TSI_*`（包括 `SQL_TSI_MONTH`、`SQL_TSI_WEEK`、`SQL_TSI_DAY`、`SQL_TSI_HOUR`、`SQL_TSI_MINUTE` 和 `SQL_TSI_SECOND`，但不包括 `SQL_TSI_YEAR`）。
\ No newline at end of file
++ 不支持 `SQL_TSI_*`（包括 `SQL_TSI_MONTH`、`SQL_TSI_WEEK`、`SQL_TSI_DAY`、`SQL_TSI_HOUR`、`SQL_TSI_MINUTE` 和 `SQL_TSI_SECOND`，但不包括 `SQL_TSI_YEAR`）。
+
+### MySQL 弃用功能导致的不兼容问题
+
+TiDB 不支持 MySQL 中标记为弃用的功能，包括：
```
Comment (ichn-hu) [inline mysql-compatibility.md:185]: TiDB 不会修复因为已经被 MySQL 标记为弃用状态的功能导致的不兼容问题，包括：

Comment (ichn-hu) [inline mysql-compatibility.md:185]:
```suggestion
TiDB 不会修复已经被 MySQL 标记为弃用状态的功能导致的不兼容问题，包括：
```

**Review by ichn-hu — CHANGES_REQUESTED** [review]

**Review by ichn-hu — APPROVED** [review]

**Pull request merged**